linear_search/linear-search.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import "fmt"
+
+func linearSearch(array []int, needle int) int {
+    for i, item := range array {
+        if item == needle {
+            return i
+        }
+    }
+    return -1
+}
+
+func main() {    
+    array := []int{0, 2, 4, 6, 8, 10, 12, 14, 16, 18}
+    index := linearSearch(array, 10)
+    if index == -1 {
+        fmt.Println("Number not found")
+    } else {
+        fmt.Println("10 found at index ", index)
```

Comment:
No need for space after "index" as different parameters in `fmt.Print` are already separated by a space when printed.